src/test/kotlin/com/exactpro/th2/check1/rule/check/TestCheckRuleTask.kt

```diff
@@ -407,40 +400,57 @@ internal class TestCheckRuleTask : AbstractCheckTaskTest() {
-            val actualLegs = verification.fields["legs"]?.fields?.values?.toList()
-            assertNotNull(actualLegs) { "Actual legs is missed" }
-
-            val expectedLegs = linkedMapOf(
-                    0 to linkedMapOf(
-                            "A" to VerificationStatus.NA,
-                            "B" to VerificationStatus.NA,
-                            "C" to VerificationStatus.FAILED,
-                            "D" to VerificationStatus.FAILED
-                            
-                    ),
-                    1 to linkedMapOf(
-                            "C" to VerificationStatus.NA,
-                            "D" to VerificationStatus.NA,
-                            "A" to VerificationStatus.FAILED,
-                            "B" to VerificationStatus.FAILED
-                    )
+            val actualLegs = assertVerificationsLegs(verificationEvent)
+
+            val expectedLegs = listOf(
+                    VerificationEntry().apply {
+                        fields = mapOf(
+                                "A" to VerificationEntry().apply { status = VerificationStatus.NA },
+                                "B" to VerificationEntry().apply { status = VerificationStatus.NA },
+                                "C" to VerificationEntry().apply { status = VerificationStatus.FAILED },
+                                "D" to VerificationEntry().apply { status = VerificationStatus.FAILED }
+                        )
+                    },
+                    VerificationEntry().apply {
+                        fields = mapOf(
+                                "C" to VerificationEntry().apply { status = VerificationStatus.NA },
+                                "D" to VerificationEntry().apply { status = VerificationStatus.NA },
+                                "A" to VerificationEntry().apply { status = VerificationStatus.FAILED },
+                                "B" to VerificationEntry().apply { status = VerificationStatus.FAILED }
+                        )
+                    }
             )
 
-            expectedLegs.forEach { (leg, verificationEntryByField) ->
-                val actualLeg = actualLegs[leg].fields
-                assertNotNull(actualLeg) { "The validation event does not contain the expected leg" }
-                verificationEntryByField.forEach { (field, status) ->
-                    val expectedVerificationEntry = actualLeg[field]
-                    assertNotNull(expectedVerificationEntry) { "Actual leg does not contain the expected field" }
-                    assertEquals(status, expectedVerificationEntry.status)
-                }
-            }
+            assertVerificationEntriesLegsByStatus(expectedLegs, actualLegs)
         })
     }
 
 
+    private fun extractVerificationFromEvent(verificationEvent: Event): Verification? {
+        return jacksonObjectMapper().readValue<List<Verification>>(verificationEvent.body.toByteArray()).firstOrNull()
```

Comment:
That won't work in case the description was added to the event's body. As a solution, we can use the `addMixIn` functionality in Jackson.
We will need to create an interface and annotate it with `@JsonSubTypes` and `@JsonTypeInfo` annotations. It will allow us to register the Verification and com.exactpro.th2.common.event.bean.Message classes  as possible options for `IBodyData`